receiver/httpcheckreceiver/scraper.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpcheckreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/httpcheckreceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/httpcheckreceiver/internal/metadata"
+)
+
+var errClientNotInit = errors.New("client not initialized")
+
+type httpcheckScraper struct {
+	client   *http.Client
+	cfg      *Config
+	settings component.TelemetrySettings
+	mb       *metadata.MetricsBuilder
+}
+
+// start starts the scraper by creating a new HTTP Client on the scraper
+func (h *httpcheckScraper) start(ctx context.Context, host component.Host) (err error) {
+	h.client, err = h.cfg.ToClient(host, h.settings)
+	return
+}
+
+// scrape connects to the endpoint and produces metrics based on the response
+func (h *httpcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	if h.client == nil {
+		return pmetric.NewMetrics(), errClientNotInit
+	}
+
+	u, err := url.Parse(h.cfg.Endpoint)
+	if err != nil {
+		return pmetric.NewMetrics(), fmt.Errorf("%s: %w", errInvalidEndpoint.Error(), err)
+	}
+
+	req := &http.Request{
+		Method: h.cfg.Method,
+		URL:    u,
+	}
+	resp, err := h.client.Do(req)
+	success := 0
+	statusCode := -1
+	if err == nil {
+		if resp.StatusCode == http.StatusOK { // change this to use configurable status codes
+			success = 1
```

Comment:
Is this comment suggesting that we'll be able to configure statuses other than 200 as successful? 👍 if so. This might be avoidable if we changed the metric representation to record the status.

Reply:
Replied to this in my response to the other comment.